src/main/java/org/zalando/baigan/service/FileSystemConfigurationRepository.java

```diff
@@ -27,22 +30,25 @@
  *
  * @author mchand
  */
-public class FileSystemConfigurationRepository extends AbstractConfigurationRepository {
+// TODO E2E test
+public class FileSystemConfigurationRepository implements ConfigurationRepository {
 
+    private static final ObjectMapper objectMapper = new ObjectMapper().registerModule(new GuavaModule());
     private static final Logger LOG = LoggerFactory.getLogger(FileSystemConfigurationRepository.class);
 
-    private final LoadingCache<String, Map<String, Configuration>> cachedConfigurations;
+    private final ConfigurationParser configurationParser;
+    private final LoadingCache<String, Map<String, Configuration<?>>> cachedConfigurations;
     private final String fileName;
 
-
-    public FileSystemConfigurationRepository(final String fileName, long refreshIntervalInSeconds) {
+    public FileSystemConfigurationRepository(final String fileName, long refreshIntervalInSeconds, final BaiganConfigClasses baiganConfigClasses) {
```

Comment:
At this point it probably makes sense to provide a builder interface equally how we provide on for the S3 repository, wdyt?

Reply:
(1) may be a separate (maybe earlier PR) (2) see also my comment regarding "who is providing the `BaiganConfigClasses` bean"

Reply:
Addressed in https://github.com/zalando-stups/baigan-config/pull/79